### PR TITLE
http: add early-data upon receiving 0-RRT request

### LINKED ISSUE #18868: quic: Envoy needs to set the early-data: 1 header when forwarding requests that were received over 0-RTT
[Section 5.1 of RFC 8470](https://datatracker.ietf.org/doc/html/rfc8470#section-5.1) specifies:

>    An intermediary that forwards a request prior to the completion of
>    the TLS handshake with its client MUST send it with the Early-Data
>    header field set to "1" (i.e., it adds it if not present in the
>    request).  

We need to make sure we do this.

### PR DESCRIPTION
**Pull request by danzh2010**
Signed-off-by: Dan Zhang <danzh@google.com>

Commit Message: Make HCM to add `Early-Data:1` header to a request sent as early data according to https://datatracker.ietf.org/doc/html/rfc8470

Risk Level: low
Testing: modified quic integration tests to test the presence of that header in upstream.
Docs Changes: none
Release Notes: none
Platform Specific Features: none
Part of #18802
Fixes #18868

### CODE REVIEW

File: test/integration/quic_http_integration_test.cc [hunk 1]
```diff
@@ -414,6 +425,18 @@ TEST_P(QuicHttpIntegrationTest, ZeroRtt) {
   }
 
   test_server_->waitForCounterEq("http3.quic_version_rfc_v1", 2u);
+
+  // Start a third connection.
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply test/integration/quic_http_integration_test.cc:429]: it'd be nice to comment that but optional( if you don't want to do another full push)

**Review by alyssawilk — APPROVED** [review]
LGTM modulo CI failure

File: test/integration/quic_http_integration_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (danzh2010) [reply test/integration/quic_http_integration_test.cc]: done

**Review by alyssawilk — APPROVED** [review]

**Comment by danzh2010** [comment]
Ping @snowp

**Review by snowp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**